### PR TITLE
[WIP] Refactor output

### PR DESCRIPTION
**Pull request by SpacePossum**
I'm targetting master here (and set [WIP]) because I would like to add `format` and `diif` to the `config interface`, which will be a big BC break. With that in place the `configuration` can be passed around without the need for passing these two additional input parameters as well (since these would be in the `configuration` then).
With the `output` param in the `configuration` the `configuration resolver` can given the `output writer` to the `fixer command`, which allows for people to add a custom `output writer` in their configuration file.
I've changed the output itself as well for some writes, for example the errors are no written in xml format in the `xml output writer` and no longer in txt.

Waiting for:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1102
That one will allow for more clean error writing/logic and interface definition for the `output writer`.

Will fix the code itself and squash all commits and the end. I'm hoping for feedback about the setup before moving on. Therefor I created this PR early as to prevent a lot rework (so this PR is not intended to be ready as is right now).

### CODE REVIEW

File: Symfony/CS/Console/Command/FixCommand.php [hunk 1]
```diff
@@ -21,7 +21,10 @@
 use Symfony\CS\Config\Config;
 use Symfony\CS\ConfigInterface;
 use Symfony\CS\Console\ConfigurationResolver;
+use Symfony\CS\Console\Output\JsonOutput;
 use Symfony\CS\Console\Output\ProcessOutput;
+use Symfony\CS\Console\Output\TxtOutput;
+use Symfony\CS\Console\Output\XmlOutput;
```
Comment (keradus) [inline Symfony/CS/Console/Command/FixCommand.php]: It looks like we have `txt` format, `xml` format, `json` format and... `process` format :D

File: Symfony/CS/Console/Output/FixerOutputInterface.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * @param bool            $isDryRun Whether the changes are simulated or not
+     * @param bool            $diff     Whether to write a diff or not
+     */
+    public function __construct(OutputInterface $output, ConfigInterface $config, $isDryRun, $diff);
+
+    public function writeChanges(array $changes);
+
+    public function writeErrors(array $errors);
+
+    public function writeInfo($info);
```
Comment (keradus) [inline Symfony/CS/Console/Output/FixerOutputInterface.php]: Tricky interface. Write many changes, many errors, but only one info.

**Comment by keradus** [comment]
> Waiting for #1102

#1102 has been merged

**Comment by SpacePossum** [comment]
thanks for the head up, I'll will start merging and rework this PR :)

**Comment by keradus** [comment]
Please rebase instead of merge ;)

**Comment by keradus** [comment]
ping @SpacePossum ;)

**Comment by keradus** [comment]
ping @SpacePossum ;)

**Comment by keradus** [comment]
What is the status of this PR @SpacePossum ?
It would be great to have it done and marged into 2.x line

**Comment by SpacePossum** [comment]
Sorry to keep you waiting; I've been full of good intentions to get working on this.
I'll pick up this soon with a rebase and new proposals. At worst you have new code changes next Wednesday .
Do you a roadmap/schedule in mind for the 2.0?

**Comment by keradus** [comment]
No I do not, there are still few major things to be done and I simply can't do it all by myself. As soon as we get all of them addressed we will release 2.0.
Please see: https://github.com/FriendsOfPHP/PHP-CS-Fixer/milestones/v2.0

**Comment by SpacePossum** [comment]
Thanks for the link :)
Looks like there is gonna be some real nice stuff in 2.0!
I asked so I could judge better if I could still make it (this PR) for 2.0, not to pressure for 2.0 or anything :)

**Comment by SpacePossum** [comment]
reworking this now, will re-open if the code is a bit more in shape.

**Comment by SpacePossum** [comment]
work continued on https://github.com/SpacePossum/PHP-CS-Fixer/commits/refactor_output_2

**Pull request closed without merging**